crates/ln-dlc-node/src/node/mod.rs

```diff
@@ -538,15 +492,79 @@ where
             storage: node_storage,
             fee_rate_estimator,
             ldk_config,
-            _background_processor_handle: background_processor_handle,
-            _connection_manager_handle: connection_manager_handle,
-            _broadcast_node_announcement_handle: broadcast_node_announcement_handle,
-            _sub_channel_manager_periodic_check_handle: sub_channel_manager_periodic_check_handle,
             network_graph,
             settings,
+            listen_address,
+            gossip_sync,
+            persister,
+            alias: alias.to_string(),
+            announcement_addresses,
+            scorer,
+            esplora_server_url,
         })
     }
 
+    /// Starts the background handles - if the returned handles are dropped, the
+    /// background tasks are stopped.
+    pub fn start(&self, event_sender: Option<watch::Sender<Option<Event>>>) -> Result<RunningNode> {
```

Comment:
I like this pattern too and we could consider it in the future! 

I'm not sure whether we actually want to have a *single* Node.

The other way to take this further would be to have separate functions instead of single `start()`
 `start_app_node(Arc<Node>, AppEventHandler) -> RunningNode`

which is more in line with your other comment I think.